src/cmd/tools/read_index_segments/main/main.go

```diff
@@ -87,34 +109,56 @@ func run(opts runOptions) {
 	// Get all fileset files.
 	log.Info("discovered namespaces", zap.Strings("namespaces", namespaces))
 
-	out, err := os.Create(opts.outputFilePath)
-	if err != nil {
-		log.Fatal("unable to create output file",
-			zap.String("file", opts.outputFilePath),
-			zap.Error(err))
+	var (
+		out                io.Writer
+		validateWorkerPool xsync.WorkerPool
+	)
+	if !opts.validate {
+		// Only validating, output to dev null.
+		out = ioutil.Discard
+		validateWorkerPool = xsync.NewWorkerPool(opts.validateConcurrency)
+		validateWorkerPool.Init()
+		log.Info("validating segment files",
+			zap.Int("concurrency", opts.validateConcurrency))
```

Comment:
Do we not need the pool when writing to file?

Reply:
Correct, no parallelism for writing to file (would cause issues).